src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -699,19 +700,26 @@ private String encodeMdcFields(AuthorizationRequestData authorizationData) {
 
     private Either<RedirectConfigurationException, URI> getRedirectUrlForPsp(
                                                                              String pspId,
-                                                                             String paymentTypeCode
+                                                                             String paymentTypeCode,
+                                                                             String channelCode
     ) {
-        String urlKey = "%s-%s".formatted(pspId, paymentTypeCode);
-        if (redirectBeApiCallUriMap.containsKey(urlKey)) {
-            return Either.right(redirectBeApiCallUriMap.get(urlKey));
-        } else {
-            return Either.left(
-                    new RedirectConfigurationException(
-                            "Missing key for redirect return url with key: [%s]".formatted(urlKey),
-                            RedirectConfigurationType.BACKEND_URLS
-                    )
-            );
+
+        return searchRedirectUrlForPsp(channelCode, pspId, paymentTypeCode);
+    }
+
+    private Either<RedirectConfigurationException, URI> searchRedirectUrlForPsp(String... params) {
+        String key = String.join("-", params);
+        if (redirectBeApiCallUriMap.containsKey(key))
+            return Either.right(redirectBeApiCallUriMap.get(key));
```

Comment:
```suggestion
        if (redirectBeApiCallUriMap.containsKey(key)) {
            return Either.right(redirectBeApiCallUriMap.get(key));
        }
            
```

Add curly brackets here

Reply:
Done.